File: hello-dyn-dart/lib/calculator.dart
// 计算器抽象基类
abstract class Calculator {
  double addFn(double a, double b);
  double subFn(double a, double b);
  double mulFn(double a, double b);
  double divFn(double a, double b);
}

// 计算器异常类
class CalculatorException implements Exception {
  final String message;
  
  const CalculatorException(this.message);
  
  @override
  String toString() => 'CalculatorException: $message';
}

File: hello-dyn-kotlin/src/main/kotlin/Calculator.kt
// 计算器接口
interface Calculator {
    fun addFn(a: Double, b: Double): Double
    fun subFn(a: Double, b: Double): Double
    fun mulFn(a: Double, b: Double): Double
    fun divFn(a: Double, b: Double): Double
}

// 计算器异常类
class CalculatorException(message: String) : Exception(message)
